releasenotes/notes/deprecate-classical-function-3bf44ef26d984366.yaml

```diff
@@ -0,0 +1,12 @@
+---
+deprecations_circuits:
+  - |
+    The :mod:`qiskit.circuit.classicalfunction` and with it
+    the :class:`.ClassicalFunction`, its related :func:`.classical_function`
+    and :class:`.BooleanExpression` have been deprecated as of Qiskit 1.4 
+    and will be removed in Qiskit 2.0. This change is done to avoid a 
+    dependency on the external library `tweedledum`, which is no longer 
+    compatible with all of Qiskit's supported platforms and Python versions.
+    For a similar functionality please use the :class:`.PhaseOracle` which
+    is going to have a `tweedledum` independent implementation, and
+    the :class:`.BitFlipOracle` which will be added in Qiskit 2.0.
```

Comment:
We could also add a code snippet as example if we want to be extra nice to the users 😄

Reply:
I second that, and maybe the release notes would be a perfect place to include an example on how to convert the phase oracle to the bit-flip oracle.